gpu-aware-scheduling/docs/usage.md

```diff
@@ -0,0 +1,115 @@
+````
+command: ["/usr/local/bin/intel_gpu_device_plugin", "-shared-dev-num", "300", "-v", "2", "-resource-manager", "true"]
+````
+
+There is a more relaxed RBAC-rule needed also as the plugin will need to read cluster POD information to get the resource amounts and annotations. You need to have this permission for the plugin:
+
+````
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["list"]
+````
+
+The GPU plugin initcontainer needs to be used in order to get the extended resources created with NFD. The initcontainer installs the required NFD-hook into the host system.
+
+## NFD
+The latest upstream NFD will be sufficient. You can use it to publish the GPU extended resources and GPU-related labels printed by the hook installed by the GPU-plugin initcontainer.
+
+For picking up the labels printed by the hook installed by the GPU-plugin initcontainer, deploy nfd master with this kind of command in its yaml:
+````
+command: ["nfd-master", "--resource-labels=gpu.intel.com/memory.max,gpu.intel.com/millicores", "--extra-label-ns=gpu.intel.com"]
+````
+
+The above would promote two labels, "memory.max" and "millicores" to extended resources of the node that produces the labels.
+
+If you want to enable i915 capability scanning, the nfd worker needs to read debugfs, and therefore it needs to run as privileged, like this:
+````
+          securityContext:
+            runAsNonRoot: null
+            # Adding GPU info labels needs debugfs "915_capabilities" access
+            # (can't just have mount for that specific file because all hosts don't have i915)
+            runAsUser: 0
+````
+
+In order to allow NFD to create extended resource, you will have to give it RBAC-rule to access nodes/status, like:
+````
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+# when using command line flag --resource-labels to create extended resources
+# you will need to uncomment "- nodes/status"
+  - nodes/status
+````
+
+## Cluster nodes
+
+You need some i915 GPUs in the nodes. Internal GPUs work fine for testing GAS, most recent NUCs are good enough.
+
+## PODs
+
+Your PODs then, needs to ask for some GPU-resources. Like this:
+````
+        resources:
+          limits:
+            gpu.intel.com/i915: 1
+            gpu.intel.com/millicores: 10
+            gpu.intel.com/memory.max: 10M
+````
+
+## Summary in a chronological order
+
+- GPU-plugin initcontainer installs an NFD hook which prints labels for you, based on the Intel GPUs it finds
+- NFD creates extended resources for you, and labels the nodes, based on the labels the hook prints
+- Your POD specs must include resource limits for GPU
+- GAS filters out nodes from deployments when necessary, and it annotates PODs
+- GPU-plugin reads annotations from the PODs and selects the GPU based on those
+
+## Some troubleshooting tips
```

Comment:
I think this guide could use an overall reference set-up / setup script. There's a lot of moving parts - GAS + Device Plugin + NFD and a lot of room for something to go wrong. Maybe have a folder in deploy that has all the components with some basic setup to run a quick scheduling workload without needing to edit yamls.

Reply:
sounds like a good idea